pkg/slice/slice.go

```diff
@@ -31,6 +36,24 @@ func RemoveString(slice []string, s string) []string {
 	return newSlice
 }
 
+// RemoveGatewayListener returns a newly created []gatewayv1alpha1.Listener
+// that contains all items from slice that are not equal to l.
+func RemoveGatewayListener(slice []gatewayv1alpha1.Listener, l gatewayv1alpha1.Listener) []gatewayv1alpha1.Listener {
+	newSlice := make([]gatewayv1alpha1.Listener, 0)
+	for _, item := range slice {
+		if apiequality.Semantic.DeepEqual(item, l) {
+			continue
+		}
+		newSlice = append(newSlice, item)
+	}
+	if len(newSlice) == 0 {
+		// Sanitize for unit tests so we don't need to distinguish empty array
+		// and nil.
+		newSlice = nil
+	}
```

Comment:
I think if you change L42 to just `var newSlice []gatewayv1alpha1.Listener` then you don't need this.